proto/regen/data/v1alpha1/events.proto

```diff
@@ -0,0 +1,32 @@
+syntax = "proto3";
+
+package regen.data.v1alpha1;
+
+option go_package = "github.com/regen-network/regen-ledger/x/data";
+
+// EventAnchorData is an event emitted when data is anchored on-chain.
+message EventAnchorData {
```

Comment:
it seems to me that these 3 events would be emitted exactly when their corresponding `Msg{*}Response` are emitted, which brings the question: why have both?

Reply:
Well they're not quite the same. These are actually a combination of the `Request` and `Response` types first of all. Also, `EventAnchorData` is _also_ emitted when `Sign` and `Store` are called if the data was not already anchored.